lib/utils/certs.go

```diff
@@ -177,25 +177,15 @@ func VerifyCertificateChain(certificateChain []*x509.Certificate) error {
 	return nil
 }
 
-// IsSelfSigned checks if the certificate is a self-signed certificate. To
-// check if a certificate is self-signed, we make sure that only one
-// certificate is in the chain and that the SubjectKeyId and AuthorityKeyId
-// match.
-//
-// From RFC5280: https://tools.ietf.org/html/rfc5280#section-4.2.1.1
-//
-//	The signature on a self-signed certificate is generated with the private
-//	key associated with the certificate's subject public key. (This
-//	proves that the issuer possesses both the public and private keys.)
-//	In this case, the subject and authority key identifiers would be
-//	identical, but only the subject key identifier is needed for
-//	certification path building.
+// IsSelfSigned checks if the certificate is a self-signed certificate. To check
+// if a certificate is self-signed, we make sure that only one certificate is in
+// the chain and that its Subject and Issuer match.
 func IsSelfSigned(certificateChain []*x509.Certificate) bool {
 	if len(certificateChain) != 1 {
 		return false
 	}
 
-	return bytes.Equal(certificateChain[0].SubjectKeyId, certificateChain[0].AuthorityKeyId)
+	return bytes.Equal(certificateChain[0].RawSubject, certificateChain[0].RawIssuer)
```

Comment:
Are we sure backporting this wont cause issue? The change itself looks good based on its how it supposed to be. But is there a chacne this might break existing cluster that "went out of the way" to create certificate that has different issuer/subject name but matching SKID and AKID?

Reply:
`IsSelfSigned` is only used to log a warning against using a self-signed webpki certificate for the proxy web listener in production, and to skip an ill-advised check at startup for the validity of the optional TLS server certs presented by the barely used metrics service. In the former case, we will catch some self-signed certificates that we are currently not noticing, and in the latter case we are actually going to be relaxing the check to allow explicitly self-signed certs even if they don't carry an AKID.

A key id collision with different name+key pairs is never intended to be possible but also not really a problem if the SKID/AKID are only considered as a hint as they should be. Either way, not something we should be touching to check if a certificate is self-signed or not.